src/components/alarm/AlarmController.cpp

```diff
@@ -88,21 +103,66 @@ uint32_t AlarmController::SecondsToAlarm() {
 
 void AlarmController::DisableAlarm() {
   xTimerStop(alarmTimer, 0);
-  state = AlarmState::Not_Set;
+  isAlerting = false;
+  if (alarm.isEnabled) {
+    alarm.isEnabled = false;
+    SaveAlarmToFile();
+  }
 }
 
 void AlarmController::SetOffAlarmNow() {
-  state = AlarmState::Alerting;
+  isAlerting = true;
   systemTask->PushMessage(System::Messages::SetOffAlarm);
 }
 
 void AlarmController::StopAlerting() {
-  // Alarm state is off unless this is a recurring alarm
-  if (recurrence == RecurType::None) {
-    state = AlarmState::Not_Set;
+  isAlerting = false;
+  // Disable alarm unless it is recurring
+  if (alarm.recurrence == RecurType::None) {
+    alarm.isEnabled = false;
+    SaveAlarmToFile();
   } else {
     // set next instance
     ScheduleAlarm();
   }
   systemTask->PushMessage(System::Messages::StopRinging);
 }
+
+void AlarmController::SetRecurrence(RecurType recurrence) {
+  if (alarm.recurrence != recurrence) {
+    alarm.recurrence = recurrence;
+    SaveAlarmToFile();
+  }
+}
+
+void AlarmController::LoadAlarmFromFile() {
+  lfs_file_t file;
+  AlarmData buffer;
+
+  if (fs.FileOpen(&file, "/alarm.dat", LFS_O_RDONLY) != LFS_ERR_OK) {
+    NRF_LOG_WARNING("[AlarmController] Failed to open alarm data file");
+    return;
+  }
+
+  fs.FileRead(&file, reinterpret_cast<uint8_t*>(&buffer), sizeof buffer);
+  fs.FileClose(&file);
+  if (buffer.version != alarmFormatVersion) {
+    NRF_LOG_WARNING("[AlarmController] Loaded alarm data has version %u instead of %u, discarding", buffer.version, alarmFormatVersion);
+    return;
+  }
+
+  alarm = buffer;
+  NRF_LOG_INFO("[AlarmController] Loaded alarm data from file");
+}
+
+void AlarmController::SaveAlarmToFile() {
+  lfs_file_t file;
+  if (fs.FileOpen(&file, "/alarm.dat", LFS_O_WRONLY | LFS_O_CREAT) != LFS_ERR_OK) {
+    NRF_LOG_WARNING("[AlarmController] Failed to open alarm data file for saving");
+    return;
+  }
+
+  fs.FileWrite(&file, reinterpret_cast<uint8_t*>(&alarm), sizeof alarm);
```

Comment:
never have seen `sizeof` as just a keyword without the braces
```suggestion
  fs.FileWrite(&file, reinterpret_cast<uint8_t*>(&alarm), sizeof(alarm));
```

Reply:
It is valid without the parentheses, The only place where they are needed is for types (`sizeof(int)` for example).
That said I personally do prefer having parentheses.

Reply:
Fixed. I checked and the [Linux kernel coding style](https://www.kernel.org/doc/html/v4.18/process/coding-style.html#spaces) and [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#sizeof) both recommend having parenthesis.